pypit/spectrographs/keck_nires.py

```diff
@@ -0,0 +1,79 @@
+""" Module for Keck/NIRES specific codes
+"""
+from __future__ import absolute_import, division, print_function
+
+try:
+    basestring
+except NameError:  # For Python 3
+    basestring = str
+
+import glob
+
+import numpy as np
+from astropy.io import fits
+
+from pypit import msgs
+from pypit import ardebug as debugger
+from. import spectroclass
+
+class KeckNIRESpectrograph(spectroclass.Spectrograph):
+    """
+    Child to handle Keck/NIRES specific code
+    """
+
+    def __init__(self):
+
+        # Get it started
+        spectroclass.Spectrograph.__init__(self)
+        self.spectrograph = 'keck_nires'
+
+    def bpm(self, shape=None, **null_kwargs):
+        """ Generate a BPM
+
+        Parameters
+        ----------
+        shape : tuple, REQUIRED
+
+        Returns
+        -------
+        badpix : ndarray
+
+        """
+        # Edges of the detector are junk
+        msgs.info("Custom bad pixel mask for NIRES")
+        self.bpm = np.zeros((self.shape[0], self.shape[1]))
```

Comment:
I think you are missing 

```
self.shape = shape
```

Reply:
Unless the tests are failing, I wouldn't worry about it for now.  I've made very significant changes to bpm.  I've gotten it such that `BPMImage` is now instrument ignorant, with a little hoop jumping.